src/api/audits.coffee

```diff
@@ -0,0 +1,169 @@
+
+    if filters['participantObjectIdentification.participantObjectID']
+      # filter by AND on same property for patientID and objectID
+      if filters['participantObjectIdentification.participantObjectID'].type
+        patientID = new RegExp filters['participantObjectIdentification.participantObjectID'].patientID
+        objectID = new RegExp filters['participantObjectIdentification.participantObjectID'].objectID
+        filters['$and'] = [ { 'participantObjectIdentification.participantObjectID': patientID }, { 'participantObjectIdentification.participantObjectID': objectID } ]
+        # remove participantObjectIdentification.participantObjectID property as we create a new '$and' operator
+        delete filters['participantObjectIdentification.participantObjectID']
+      else
+        participantObjectID = JSON.parse filters['participantObjectIdentification.participantObjectID']
+        filters['participantObjectIdentification.participantObjectID'] = new RegExp "#{participantObjectID}"
+
+    # execute the query
+    this.body = yield Audit
+      .find filters, projectionFiltersObject
+      .skip filterSkip
+      .limit filterLimit
+      .sort 'eventIdentification.eventDateTime': -1
+      .exec()
+  catch e
+    utils.logAndSetResponse this, 'internal server error', "Could not retrieve audits via the API: #{e}", 'error'
+
+
+
+
+
+###
+# Retrieves the details for a specific Audit Record
+###
+exports.getAuditById = (auditId) ->
+  # Must be admin
+  if not authorisation.inGroup 'admin', this.authenticated
+    utils.logAndSetResponse this, 'forbidden', "User #{this.authenticated.email} is not an admin, API access to getAuditById denied.", 'info'
+    return
+
+  # Get the values to use
+  auditId = unescape auditId
+
+  try
+    # get projection object
+    projectionFiltersObject = getProjectionObject 'full'
+
+    result = yield Audit.findById(auditId, projectionFiltersObject).exec()
+
+    # Test if the result if valid
+    if not result
+      this.body = "Could not find audits record with ID: #{auditId}"
+      this.status = 'not found'
+    else
+      this.body = result
+
+  catch e
+    utils.logAndSetResponse this, 'internal server error', "Could not get audit by ID via the API: #{e}", 'error'
+
+
+
+###
+# construct audit filtering dropdown options
+###
+exports.getAuditsFilterOptions = ->
+
+  # Must be admin
+  if not authorisation.inGroup 'admin', this.authenticated
+    utils.logAndSetResponse this, 'forbidden', "User #{this.authenticated.email} is not an admin, API access to getAudits denied.", 'info'
+    return
+
+  try
+
+    # execute the query
+    eventID = yield Audit.distinct('eventIdentification.eventID').exec()
+    eventTypeCode = yield Audit.distinct('eventIdentification.eventTypeCode').exec()
+    roleIDCode = yield Audit.distinct('activeParticipant.roleIDCode').exec()
+    participantObjectIDTypeCode = yield Audit.distinct('participantObjectIdentification.participantObjectIDTypeCode').exec()
```

Comment:
Can we add `auditSourceId` to the filter options result as well? It would be really cool to have that as a dropdown on the frontend.